cmd/ipfs-cluster-ctl/main.go

```diff
@@ -610,6 +610,63 @@ although unpinning operations in the cluster may take longer or fail.
 						return nil
 					},
 				},
+				{
+					Name:  "update",
+					Usage: "Pin a new item based on an existing one",
+					Description: `
+This command will add a new pin to the cluster taking all the options from an
+existing one, including name. This means that the new pin will bypass the
+allocation process and will be allocated to the same peers as the existing
+one.
+
+The cluster peers will try to Pin the new item on IPFS using the "pin update"
+command. This is especially efficient when the content of two pins (their DAGs)
+are similar.
+
+Unlike the "pin update" command in the ipfs daemon, this will not unpin the
+existing item from the cluster. Please run "pin rm" for that.
```

Comment:
No. I have considered all this.

1) First, we cannot do atomic Pin/Rm after PinAdd (update) operation. Doing a non-atomic pin rm opens the door to leaving the system in inconsistent state. If the unpin operation fails, you have an added pin in there. This could be done as a CLI feature (where --unpin is a shortcut for running pin update + unpin), BUT
2) In Cluster, pinning is an async operation. Pinning un cluster does not mean that the content is pinned in IPFS. Unpinning after pinning will likely unpin content from IPFS while the pinning process assocaited to pin/update has not completed. This can get to ugly places it it happens during GC.

> I would be cleaner to have a separate api endpoint for pin/update

A pin/update is just a pin/add: it creates a pin resource (or an allocation). The extra endpoint means adding extra code for an non-restful endpoint. the `/pins/<cid>/update/<newcid>` is also weird because, for the reasons above, the original pin is not modified.

>  Plus it carries api.Pin unnecessarily. We only need to and from

We cannot call an RPC endpoint with two IN arguments. We need to wrap them in a single object. We use the Pin for that because that's the right object for creating a new pin and for general consistency, given the fact that it calls the Pin rpc endpoint. Again, you're asking for more boilerplate code for 0 gain. The approach here is the iteration upon all those ideas into the most consistent/simple approach.

>  since we are getting options from PinGet

For now, we don't know if in the future we may extend options. Useful feedback here would have been to realize that we should probably respect the `Name` option for the update-pin and not use the one from the old pin. I will change that.

Reply:
Also, note that the requesters of this feature use it with --unpin=false, meaning unpin is not always the natural step even though i thought the same.